spring-cloud-huawei-nacos/spring-cloud-huawei-nacos-config/src/main/java/com/huaweicloud/nacos/config/client/NacosPropertySourceExtendLocator.java

```diff
@@ -0,0 +1,110 @@
+/*
+
+ * Copyright (C) 2020-2024 Huawei Technologies Co., Ltd. All rights reserved.
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.huaweicloud.nacos.config.client;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import org.apache.commons.lang3.StringUtils;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.web.client.RestTemplate;
+
+import com.alibaba.cloud.nacos.NacosConfigProperties;
+import com.alibaba.nacos.shaded.com.google.gson.Gson;
+import com.alibaba.nacos.shaded.com.google.gson.reflect.TypeToken;
+import com.huaweicloud.nacos.config.NacosConfigConst;
+import com.huaweicloud.nacos.config.manager.ConfigServiceManagerUtils;
+
+public class NacosPropertySourceExtendLocator {
+  private static final Logger LOGGER = LoggerFactory.getLogger(NacosPropertySourceExtendLocator.class);
+
+  private static final String NACOS_CONFIG_QUERY_URI
+      = "%s/nacos/v1/cs/configs?dataId=%s*&group=%s&pageNo=1&pageSize=1000&tenant=%s&search=blur";
+
+  private final TypeToken<List<PropertyConfigItem>> typeToken = new TypeToken<List<PropertyConfigItem>>() {};
+
+  private final NacosConfigProperties properties;
+
+  public NacosPropertySourceExtendLocator(NacosConfigProperties nacosConfigProperties) {
+    this.properties = nacosConfigProperties;
+  }
+
+  public List<PropertyConfigItem> loadRouterProperties() {
+    try {
+      RestTemplate restTemplate = new RestTemplate();
```

Comment:
Using nacos client if possible or a lot of other configurations need to be considered for RestTemplate, you need to write a lot of network codes, like security, pool and reuse.